File: packages/javascript/lib/word/randAnimal.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Animals — the creatures a nickname is most often built around.
 *
 * `randWord({ theme: 'animal' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randAnimal({ language: 'ko', count: 3 }); // ['사자', '호랑이', '수달']
 * randAnimal({ language: 'en', count: 3 }); // ['Lion', 'Otter', 'Falcon']
 */
export function randAnimal(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Animal words with the language and theme each one came from.
 *
 * @example
 * randAnimal({ language: 'ko', output: 'detail' });
 * // [{ word: '사자', language: 'ko', theme: 'animal' }]
 */
export function randAnimal(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randAnimal(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('animal', options);
}

File: packages/dart/lib/src/word/rand_animal.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Animals — the creatures a nickname is most often built around.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.animal] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randAnimal(language: WordLanguage.ko, count: 3); // [사자, 호랑이, 수달]
/// randAnimal(language: WordLanguage.en, count: 3); // [Lion, Otter, Falcon]
/// ```
List<String> randAnimal({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.animal,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
